### ISSUE TITLE
Compatibility with RepeatModeler 2.x

### ISSUE THREAD

**Issue by kfuku52** [nextgenusfs/funannotate#600]
**Are you using the latest release?**
yes, 1.8.8.

**Describe the bug**
At the moment, `funannotate mask` does not seem to support RepeatModeler 2.x. This is because the `RepeatModeler -e` option in 1.x has been replaced by `BuildDatabase -engine`, This can be easily fixed like in this branch. 
https://github.com/kfuku52/funannotate/commit/9bad1296fc337049b87a94127f92bdd0fdeea186
(This branch also fixes a bug where --repeatmasker_species is not passed correctly to RepeatMasker.)

I'll be happy to create a PR, but the change in my branch loses compatibility with RepeatModeler 1.x. It seems possible to get version information from RepeatModeler's help messages to support both 1.x and 2.x. So if the 1.x support is still needed, I can update my branch so and create a PR. Please let me know your thought.

**Comment by hyphaltip** [comment]
I def have this bug too- I have not worked on the workaround - but if you want to provide the option to run v1 or v2 that would be better -- I am just running RM outside of funannotate for the time being but I think having this better rolled in would be helpful

**Comment by nextgenusfs** [comment]
I'm not sure I want to support RepeatModeler/Masker -- I've not upgraded the code for several years in relation to this because install is not straightforward as well as the RepBase database is no longer available, making masking (at least for fungi and other non-model organisms) difficult.  I sort of view it as out of scope with funannotate.

**Comment by kfuku52** [comment]
I don't have the RepBase subscription but RepeatMasker is still useful because the latest version comes with the out-of-the-box Dfam repeat database (see [here](http://www.repeatmasker.org/RepeatMasker/)), so Repbase is no longer necessary.

bioconda provides the recipes for both, so currently it's pretty easy to install if you don't have to manually compile it from the source code: `conda install -c bioconda repeatmasker repeatmodeler`. bioconda's latest RepeatModeler is 2.x.

**Comment by nextgenusfs** [comment]
Right, DFAM contains repeats of only 5 species.... so for my usage (fungi) it isn't helpful. 


> The current release (Dfam 3.2) contains 6,900 TE families spanning five organisms: human, mouse, zebrafish, fruit fly, nematode, and a growing number of additional species. To supplement this databases we recommend obtaining the RepeatMasker edition of RepBase.

**Comment by nextgenusfs** [comment]
Yeah, I'm not saying it isn't useful if you are annotating human, mouse, zebrafish, etc --> but most of us are trying to annotate non-model organisms so the DFAM repeat library isn't going to be very useful.  RepeatModeler used to also require RepBase library in order to do the de novo predictions, I don't know if that is still the case or not.

**Comment by nextgenusfs** [comment]
Looks like you can browse species here: https://dfam.org/browse?clade=4890&clade_descendants=true&include_raw=true

**Comment by hyphaltip** [comment]
I'm going to go ahead and try and get repeatmodeler 2.x + repeatmasker 4.1 series working so refer to those branch fixes to this bug here.
